google-cloud-bigquery/lib/google/cloud/bigquery/dataset.rb

```diff
@@ -466,6 +467,25 @@ def access
           access_builder.freeze
         end
 
+        ##
+        # Retrieves the tags associated with this dataset. Tag keys are
+        # globally unique, and managed via the resource manager API.
+        #
+        # @see https://cloud.google.com/resource-manager/docs/tags/tags-overview
+        # for more information.
+        #
+        # @return [Google::Cloud::Bigquery::Dataset::Tag] The list of tags.
+        #
+        def tags
+          ensure_full_data!
+          return nil if @gapi.tags.nil?
+          tags = []
+          @gapi.tags.each do |gapi|
+            tags.append Tag.from_gapi(gapi)
+          end
```

Comment:
You can use `map` method instead of a `loop + append`. Makes the code shorter, easier to read, and avoids mutation.

i.e; replace this: 👇
```rb
tags = []
@gapi.tags.each do |gapi|
  tags.append Tag.from_gapi(gapi)
end
tags
```
with this: 👇
```rb
tags = @gapi.tags.map { |gapi| Tag.from_gapi(gapi) }
```

Reply:
Thanks a lot!! will update the code.